app/uk/gov/hmrc/servicedependencies/controller/ServiceDependenciesController.scala

```diff
@@ -285,6 +206,111 @@ class ServiceDependenciesController @Inject()(
       ).merge
     }
 
+  private def getRepositoryInfo(meta: MetaArtefact, latestVersions: Seq[LatestVersion], bobbyRules: BobbyRules) = {
+    def toDependencies(name: String, scope: DependencyScope, dotFile: String) =
+      slugDependenciesService.curatedLibrariesFromGraph(
+        dotFile = dotFile,
+        rootName = name,
+        latestVersions = latestVersions,
+        bobbyRules = bobbyRules,
+        scope = scope
+      )
+
+    val sbtVersion =
+    // sbt-versions will be the same for all modules,
+      meta.modules.flatMap(_.sbtVersion.toSeq).headOption
+
+    def when[T](b: Boolean)(seq: => Seq[T]): Seq[T] =
+      if (b) seq
+      else Nil
+
+    def dependencyIfMissing(
+                             dependencies: Seq[Dependency],
+                             group: String,
+                             artefact: String,
+                             versions: Seq[Version],
+                             scope: DependencyScope
+                           ): Seq[Dependency] =
+      for {
+        v <- versions
+        if dependencies.find(dep => dep.group == group && dep.name == artefact).isEmpty
+      } yield
+        Dependency(
+          name = artefact,
+          group = group,
+          currentVersion = v,
+          latestVersion = latestVersions.find(d => d.name == artefact && d.group == group).map(_.version),
+          bobbyRuleViolations = List.empty,
+          importBy = None,
+          scope = Some(scope)
+        )
+
+    val buildDependencies = meta.dependencyDotBuild.fold(Seq.empty[Dependency])(s => toDependencies(meta.name, DependencyScope.Build, s))
+
+    val repository =
+      Repository(
+        name = meta.name,
```

Comment:
the formatting of this function is hard to read - can we align the field names and values?